pkg/proto/datadog/remoteconfig/remoteconfig.proto

```diff
@@ -0,0 +1,104 @@
+syntax = "proto3";
+
+package datadog.config;
+
+option go_package = "pkg/proto/pbgo";  // golang
+
+// Backend definitions
+
+message ConfigMetas {
+  repeated TopMeta roots = 1;
+  TopMeta timestamp = 2;
+  TopMeta snapshot = 3;
+  TopMeta topTargets = 4;
+  repeated DelegatedMeta delegatedTargets = 5;
+}
+
+message DirectorMetas {
+  repeated TopMeta roots = 1;
+  TopMeta timestamp = 2;
+  TopMeta snapshot = 3;
+  TopMeta targets = 4;
+}
+
+message DelegatedMeta {
+  uint64 version = 1;
+  string role = 2;
+  bytes raw = 3;
+}
+
+message TopMeta {
+  uint64 version = 1;
+  bytes raw = 2;
+}
+
+message File {
+  string path = 1;
+  bytes raw = 2;
+}
+
+// Backend queries
+
+message LatestConfigsRequest {
+  string hostname = 1;
+  string agentVersion = 2;
+  // timestamp and snapshot versions move in tandem so they are the same.
+  uint64 current_config_snapshot_version = 3;
+  uint64 current_config_root_version = 9;
+  uint64 current_director_root_version = 8;
+  repeated string products = 4;
+  repeated string new_products = 5;
+  repeated Client active_clients = 6;
+}
+
+message LatestConfigsResponse {
+  ConfigMetas config_metas = 1;
+  DirectorMetas director_metas = 2;
+  repeated File target_files = 3;
+}
+
+// Client defintions
+
+message Client {
+  ClientState state = 1;
+  string id = 2;
+  repeated string products = 3;
+  string name = 4;
+  string version = 5;
+  bool is_tracer = 6;
+  TracerClient tracer_client = 7;
+}
+
+message TracerClient {
+  string runtime_id = 1;
+  string language = 2;
+  string tracer_version = 3;
```

Comment:
I was thinking, since there might be more additional metadata that we might want to add in the future: should we have a `repeated string tags = x;` instead of adding new top-level fields to the message?

Reply:
The ones we currently have describe a tracer entirely and they make sense as top-level fields IMO. Adding a list of tag is fine to be future proof for optional meta we want to forward in the future but I don't have a strong opinion on wether or not this should be done right now. Since you're the stakeholder here I'll let you make the call :)

Reply:
Nah, I suppose it doesn't have to be now. We can add it when we need it. 👍